devices/ePaper/Drivers/Gdew0154x/Gdew0154x.cs

```diff
@@ -0,0 +1,636 @@
+        public virtual void PowerOn()
+        {
+            HardwareReset();
+            WaitMs(1000);
+            SendCommand(0x50);
+            SendData(0xd7);
+            SendCommand((byte)Command.PowerOn);
+            WaitReady();
+
+            PowerState = PowerState.PoweredOn;
+        }
+
+        /// <inheritdoc/>
+        public virtual void SendCommand(params byte[] command)
+        {
+            // make sure we are setting data/command pin to low (command mode)
+            _dataCommandPin.Write(PinValue.Low);
+
+            foreach (var b in command)
+            {
+                // write the command byte to the display controller
+                _spiDevice.WriteByte(b);
+            }
+        }
+
+        /// <inheritdoc/>
+        public virtual void SendData(params byte[] data)
+        {
+            // set the data/command pin to high to indicate to the display we will be sending data
+            _dataCommandPin.Write(PinValue.High);
+
+            foreach (var @byte in data)
+            {
+                _spiDevice.WriteByte(@byte);
+            }
+
+            // go back to low (command mode)
+            _dataCommandPin.Write(PinValue.Low);
+        }
+
+        /// <summary>
+        /// Sets the current active frame buffer page to the specified page index.
+        /// Existing frame buffer is reused by clearing it first and page bounds are recalculated.
+        /// Make sure to call <see cref="PerformFullRefresh"/> or <see cref="PerformPartialRefresh"/>
+        /// to persist the frame to the display's RAM before calling this method.
+        /// </summary>
+        /// <param name="page">The frame buffer page index to move to.</param>
+        protected virtual void SetFrameBufferPage(int page)
+        {
+            if (page < 0 || page >= PagesPerFrame)
+            {
+                page = 0;
+            }
+
+            FrameBuffer1bpp.Clear();
+            CurrentFrameBufferPage = page;
+
+            CalculateFrameBufferPageBounds();
+        }
+
+        /// <inheritdoc/>
+        public virtual void SetPosition(int x, int y)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Performs the software reset commands sequence on the display.
+        /// </summary>
+        protected virtual void SoftwareReset()
+        {
+            SendCommand(0x04);
+
+            WaitReady(PinValue.High);
+            WaitMs(10);
+        }
+
+        /// <summary>
+        /// A simple wait method that wraps <see cref="Thread.Sleep(int)"/>.
+        /// </summary>
+        /// <param name="milliseconds">Number of milliseconds to sleep.</param>
+        protected virtual void WaitMs(int milliseconds)
+        {
+            Thread.Sleep(milliseconds);
+        }
+
+        /// <inheritdoc/>
+        public virtual void WaitReady(PinValue pinValue)
+        {
+            while (_busyPin.Read() != pinValue)
+            {
+                WaitMs(5);
```

Comment:
if waiting is a matter of few milliseconds you can use the Thread.Spin. If it's indeed much more, then keep the Thread.Sleep